dev/devtest/testutils.go

```diff
@@ -110,28 +120,34 @@ func WithStdoutCaptured(f func()) string {
 
 // RemoveDirectoryOrFail removes a directory (and its contents) or fails the test.
 func RemoveDirectoryOrFail(t *testing.T, path string) {
+	if !DirExists(path) {
+		assert.FailNow(t, fmt.Sprintf("%s does not exist", path))
+	}
+
 	if err := os.RemoveAll(path); err != nil {
 		assert.Fail(t, err.Error())
+	} else {
+		LogDebugf("removed %s\n", path)
 	}
 }
 
 // MakeDirectoryOrFail makes a directory or fails the test, returning the path
 // of the directory that was created.
-func MakeDirectoryOrFail(t *testing.T, path string) string {
-	if err := os.Mkdir(path, 0755); err != nil {
+func MakeDirectoryOrFail(t *testing.T, path string) func() {
+	if err := os.MkdirAll(path, 0755); err != nil {
 		assert.Fail(t, err.Error())
 	}
-	return path
+
+	return func() {
+		RemoveDirectoryOrFail(t, path)
+	}
 }
 
 // WithWorkingDirectory executes the passed function within the context of
-// the passed working directory path. If the directory does not exist,
-// WithWorkingDirectory will attempt to create it.
+// the passed working directory path. If the directory does not exist, panic.
 func WithWorkingDirectory(path string, f func()) {
 	if _, err := os.Stat(path); os.IsNotExist(err) {
-		if err := os.MkdirAll(path, 0755); err != nil {
-			panic(err)
-		}
+		panic(err)
```

Comment:
it's more complicated to have this helper make the path if it doesn't exist, so just bail out if you didn't create the directory in the first place.